packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2231,6 +2235,8 @@ class RenderCustomPaint extends RenderProxyBox {
     super.paint(context, offset);
     if (_foregroundPainter != null)
       _paintWithPainter(context.canvas, offset, _foregroundPainter);
+    if (forceRasterCache)
+      context.setIsComplexHint();
```

Comment:
you should also set this before calling super.paint, in case the background and foreground painters end up painting on different layers.